File: linkedin-frontend/src/app/model/UserProfile.ts
export enum UserRole {
  ADMIN = "ADMIN",
  USER = "USER"
}

export class UserProfile {
  private _chatId: number;
  private _username: string;
  private _botState: string;
  private _registeredAt: Date;
  private _userRole: UserRole;
  private _userPic: string;


  constructor(chatId: number, username: string, botState: string, registeredAt: Date, userRole: UserRole, userPic: string) {
    this._chatId = chatId;
    this._username = username;
    this._botState = botState;
    this._registeredAt = registeredAt;
    this._userRole = userRole;
    this._userPic = userPic;
  }


    get chatId(): number {
    return this._chatId;
  }

  set chatId(value: number) {
    this._chatId = value;
  }

  get username(): string {
    return this._username;
  }

  set username(value: string) {
    this._username = value;
  }

  get botState(): string {
    return this._botState;
  }

  set botState(value: string) {
    this._botState = value;
  }

  get registeredAt(): Date {
    return this._registeredAt;
  }

  set registeredAt(value: Date) {
    this._registeredAt = value;
  }

  get userRole(): UserRole {
    return this._userRole;
  }

  set userRole(value: UserRole) {
    this._userRole = value;
  }

  get userPic(): string {
    return this._userPic;
  }

  set userPic(value: string) {
    this._userPic = value;
  }
}

File: modules/postgresdb/src/main/java/com/gbyzzz/linkedinjobsbot/modules/postgresdb/entity/UserProfile.java
package com.gbyzzz.linkedinjobsbot.modules.postgresdb.entity;

import com.gbyzzz.linkedinjobsbot.modules.postgresdb.entity.type.PGUserProfileBotState;
import com.gbyzzz.linkedinjobsbot.modules.postgresdb.entity.type.PGUserProfileUserRole;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.Type;
import org.hibernate.proxy.HibernateProxy;

import java.util.Date;
import java.util.Objects;

@Entity
@Getter
@Setter
@ToString
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "user_profiles")
public class UserProfile {
    @Id
    private Long chatId;
    @Column(name = "username")
    private String username;
    @Column(name = "bot_state")
    @Type(PGUserProfileBotState.class)
    @Enumerated(EnumType.STRING)
    private BotState botState;
    @Column(name = "registered_at")
    private Date registeredAt;
    private String password;
    @Type(PGUserProfileUserRole.class)
    private UserRole userRole;
    private String userPic;

    public enum UserRole {
        ADMIN,
        USER
    }

    public enum BotState {
        NA,
        ADD_KEYWORDS,
        ADD_LOCATION,
        ADD_EXPERIENCE,
        ADD_JOB_TYPE,
        ADD_WORKPLACE,
        ADD_FILTERS,
        ADD_FILTER_INCLUDE,
        ADD_FILTER_EXCLUDE,
        ADD_FILTER_JOB_TYPE,
        ADD_FILTER_WORKPLACE,
        NEW,
        APPLIED,
        SEARCHES
    }

    @Override
    public final boolean equals(Object o) {
        if (this == o) return true;
        if (o == null) return false;
        Class<?> oEffectiveClass = o instanceof HibernateProxy ? ((HibernateProxy) o).getHibernateLazyInitializer().getPersistentClass() : o.getClass();
        Class<?> thisEffectiveClass = this instanceof HibernateProxy ? ((HibernateProxy) this).getHibernateLazyInitializer().getPersistentClass() : this.getClass();
        if (thisEffectiveClass != oEffectiveClass) return false;
        UserProfile that = (UserProfile) o;
        return getChatId() != null && Objects.equals(getChatId(), that.getChatId());
    }

    @Override
    public final int hashCode() {
        return getClass().hashCode();
    }
}

